code/_macros.dm

```diff
@@ -97,19 +109,24 @@
 
 #define random_id(key,min_id,max_id) uniqueness_repository.Generate(/datum/uniqueness_generator/id_random, key, min_id, max_id)
 
+/proc/place_meta_charset(content)
+	if(istext(content))
+		content = "<meta charset=\"utf-8\">" + content
+	return content
+
 #define to_chat(target, message)                            target << (message)
 #define to_world(message)                                   world << (message)
 #define to_world_log(message)                               world.log << (message)
 #define sound_to(target, sound)                             target << (sound)
 #define to_file(file_entry, source_var)                     file_entry << (source_var)
 #define from_file(file_entry, target_var)                   file_entry >> (target_var)
-#define show_browser(target, browser_content, browser_name) target << browse(browser_content, browser_name)
+#define show_browser(target, browser_content, browser_name) target << browse(place_meta_charset(browser_content), browser_name)
```

Comment:
I don't really like this solution given that it generates invalid HTML, but I get why it's done - would be better to have a more general proc that adds `<!DOCTYPE html>` and `<html>` around a given body at some point.

@MistakeNot4892 Thoughts?

Reply:
I am not overly familiar with HTML beyond formatting, scripts and tables, I don't really have an opinion on this.